spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -298,6 +317,97 @@ public void deleteWhere(Filter[] filters) {
     }
   }
 
+  @Override
+  public StructType partitionSchema() {
+    List<Types.NestedField> fields = icebergTable.spec().partitionType().fields();
```

Comment:
This should use the partition metadata table's Iceberg schema and convert it to Spark using `SparkSchemaUtil.convert`. Also, it would be great to reuse the value using a lazily initialized instance variable.